spec.md

```diff
@@ -427,6 +437,83 @@ Ext format family stores a tuple of an integer and a byte array.
     * type < 0 is reserved for future extension including 2-byte type information
 
 
+<a name="formats-timestamp"/>
+### Timestamp extension type
+
+Timestamp extension type is assigned to extension type `-1`. It defines 3 formats: 32-bit format, 64-bit format, and 96-bit format.
+
+    timestamp 32 stores the number of seconds that have elapsed since 1970-01-01 00:00:00 UTC
+    in an 32-bit unsigned integer:
+    +--------+--------+--------+--------+--------+--------+
+    |  0xd6  |   -1   |   seconds in 32-bit unsigned int  |
+    +--------+--------+--------+--------+--------+--------+
+
+    timestamp 64 stores the number of seconds and nanoseconds that have elapsed since 1970-01-01 00:00:00 UTC
+    in 2 32-bit unsigned integers:
+    +--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+    |  0xd7  |   -1   |nanoseconds in 30-bit unsigned int|   seconds in 34-bit unsigned int   |
+    +--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+
+    timestamp 96 stores the number of seconds and nanoseconds that have elapsed since 1970-01-01 00:00:00 UTC
+    in 64-bit signed integer and 32-bit unsigned integer:
+    +--------+--------+--------+--------+--------+--------+--------+
+    |  0xc7  |   12   |   -1   |nanoseconds in 32-bit unsigned int |
+    +--------+--------+--------+--------+--------+--------+--------+
+    +--------+--------+--------+--------+--------+--------+--------+--------+
+                        seconds in 64-bit signed int                        |
+    +--------+--------+--------+--------+--------+--------+--------+--------+
+
+* Timestamp 32 format can represent a timestamp in [1970-01-01 00:00:00 UTC, 2106-02-07 06:28:16 UTC) range. Nanoseconds part is 0.
+* Timestamp 64 format can represent a timestamp in [1970-01-01 00:00:00.000000000 UTC, 2514-05-30 01:53:04.000000000 UTC) range.
+* Timestamp 96 format can represent a timestamp in [-584554047284-02-23 16:59:44 UTC, 584554051223-11-09 07:00:16.000000000 UTC) range.
+* In timestamp 64 and timestamp 96 formats, nanoseconds must not be larger than 999999999.
```

Comment:
I added this restriction specifically because deserialization becomes unexpectedly complicated if nanosecond part includes carry over, and following this restriction doesn't make serialization code complicated.